hibernate-core/src/test/java/org/hibernate/test/component/empty/SelectBeforeUpdateEmbeddedTest.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.component.empty;
+
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.persistence.Embeddable;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Version;
+
+import org.hibernate.EmptyInterceptor;
+import org.hibernate.annotations.SelectBeforeUpdate;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.type.Type;
+import org.junit.Test;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.hibernate.testing.transaction.TransactionUtil;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Chris Cranford
+ */
+public class SelectBeforeUpdateEmbeddedTest extends BaseCoreFunctionalTestCase {
+	private final OnFlushDirtyInterceptor i = new OnFlushDirtyInterceptor();
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { Person.class };
+	}
+
+	@Override
+	protected void configure(Configuration configuration) {
+		super.configure( configuration );
+		// applies the OnFlushDirtyInterceptor that is meant to track whether a Flush event fires
+		// in situations where it ultimately shouldn't because null composites should equate an
+		// instantiated component with all null properties.
+		configuration.setInterceptor( i );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-11237")
+	public void testSelectBeforeUpdateUsingEmptyComposites() {
+		// Opt-in behavior 5.1+
+		rebuildSessionFactory( c -> c.setProperty( AvailableSettings.CREATE_EMPTY_COMPOSITES_ENABLED, "true" ) );
+		testSelectBeforeUpdate();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-11237")
+	public void testSelectBeforeUpdateUsingNullComposites() {
+		// Legacy behavior test
+		rebuildSessionFactory( c -> c.setProperty( AvailableSettings.CREATE_EMPTY_COMPOSITES_ENABLED, "false" ) );
+		testSelectBeforeUpdate();
+	}
+
+	/**
+	 * Performs various tests both available attached and detached entities which use
+	 * the {@code @SelectBeforeUpdate} annotation with an {code @Embedded} component.
+	 */
+	private void testSelectBeforeUpdate() {
+		rebuildSessionFactory(  );
```

Comment:
I think this line should be removed since otherwise, we rebuild the `SessionFactory` without the explicit `AvailableSettings.CREATE_EMPTY_COMPOSITES_ENABLED` setting.